src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_ObjectCreationExpression.cs

```diff
@@ -101,7 +101,35 @@ public override BoundNode VisitObjectCreationExpression(BoundObjectCreationExpre
                 return rewrittenObjectCreation;
             }
 
-            return MakeObjectCreationWithInitializer(node.Syntax, rewrittenObjectCreation, node.InitializerExpressionOpt, node.Type);
+            return MakeExpressionWithInitializer(node.Syntax, rewrittenObjectCreation, node.InitializerExpressionOpt, node.Type);
+        }
+
+        public override BoundNode VisitWithExpression(BoundWithExpression withExpr)
+        {
+            RoslynDebug.AssertNotNull(withExpr.CloneMethod);
```

Comment:
Curious: why are we using `RoslynDebug.AssertNotNull` here vs. `Debug.Assert(withExpr.CloneMethod is object)` (or insert your favorite non-null expression form).

Reply:
I think I just forgot that `debug.assert` is annotated now